pkg/codegen/nodejs/utilities.go

```diff
@@ -99,6 +99,9 @@ func makeValidIdentifier(name string) string {
 		}
 	}
 	name = builder.String()
+	if strings.HasPrefix(name, "pulumi_") {
+		name = strings.TrimPrefix(name, "pulumi_")
+	}
```

Comment:
Trim prefix includes its own existence check, so you don't need to duplicate it.
```suggestion
	name = strings.TrimPrefix(name, "pulumi_")
```

This is the implementation of `strings.TrimPrefix`:
```go
func TrimPrefix(s, prefix string) string {
	if HasPrefix(s, prefix) {
		return s[len(prefix):]
	}
	return s
}
```